lib/pub_sub/message.rb

```diff
@@ -1,7 +1,13 @@
 module PubSub
   class Message
     def initialize(payload)
-      @payload = JSON.parse(payload)
+      content = JSON.parse(payload)
+      if content.is_a?(Hash) && content.has_key?('Message')
```

Comment:
I looked into this but was not able to find a non trivial way of detecting the mode. Decided this was enough for the moment. 

Also I believe that it would require a payload with `Message` not `message`